src/main/java/org/jabref/gui/search/SearchResultsTableDataModel.java

```diff
@@ -0,0 +1,78 @@
+package org.jabref.gui.search;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.binding.Bindings;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import javafx.collections.transformation.SortedList;
+
+import org.jabref.gui.StateManager;
+import org.jabref.gui.maintable.BibEntryTableViewModel;
+import org.jabref.gui.maintable.MainTableFieldValueFormatter;
+import org.jabref.logic.search.SearchQuery;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.groups.GroupTreeNode;
+import org.jabref.preferences.PreferencesService;
+
+import com.tobiasdiez.easybind.EasyBind;
+
+public class SearchResultsTableDataModel {
+
+    private final FilteredList<BibEntryTableViewModel> entriesFiltered;
+    private final SortedList<BibEntryTableViewModel> entriesSorted;
+    private final ObjectProperty<MainTableFieldValueFormatter> fieldValueFormatter;
+    private final PreferencesService preferencesService;
+    private final BibDatabaseContext bibDatabaseContext;
+
+    public SearchResultsTableDataModel(BibDatabaseContext bibDatabaseContext, PreferencesService preferencesService, StateManager stateManager) {
+        this.preferencesService = preferencesService;
+        this.bibDatabaseContext = bibDatabaseContext;
+        this.fieldValueFormatter = new SimpleObjectProperty<>(
+                                                              new MainTableFieldValueFormatter(preferencesService, bibDatabaseContext));
+
+        List<BibEntryTableViewModel> models = new ArrayList<>();
+        for (BibDatabaseContext bibdatabasecontext : stateManager.getOpenDatabases()) {
+            ObservableList<BibEntry> entriesForDb = bibdatabasecontext.getDatabase().getEntries();
+            List<BibEntryTableViewModel> viewModelForDb = EasyBind.mapBacked(entriesForDb, entry -> new BibEntryTableViewModel(entry, bibdatabasecontext, fieldValueFormatter));
+
+            models.addAll(viewModelForDb);
+        }
+        ObservableList<BibEntryTableViewModel> entriesViewModel = FXCollections.observableArrayList(models);
```

Comment:
Why not making the models list an ObservableList from the start?

Reply:
Suprisingly it works now. Previously I was getting an exception on calling addAll on the observableArraylist...